### PR TITLE
Provide more information on error

### PR DESCRIPTION
**Pull request by hobbypunk90**

### CODE REVIEW

File: lib/omniauth/strategies/openid_connect.rb [hunk 1]
```diff
@@ -111,7 +111,7 @@ def callback_phase
         invalid_state = params['state'].to_s.empty? || params['state'] != stored_state
 
         raise CallbackError.new(params['error'], error_description, params['error_uri']) if error
-        raise CallbackError, 'Invalid state parameter' if invalid_state
+        raise CallbackError.new(:csrf_detected, "Invalid state parameter") if invalid_state
```
Comment (m0n9oose) [inline lib/omniauth/strategies/openid_connect.rb]: Please, revert this change back to old version

https://github.com/rubocop-hq/ruby-style-guide#exception-class-messages

Comment (hobbypunk90) [reply lib/omniauth/strategies/openid_connect.rb]: in most cases it is right to follow the guide, but i think in that case it isn't.
without raising the error instance it is not possible to provide more informations to the error.

**Comment by hobbypunk90** [comment]
@m0n9oose can you take a second look at this? I changed the error handling ;)

**Comment by m0n9oose** [comment]
@hobbypunk90 hey there.

Sorry for slow response, i'm quite busy these days.

I see you added nice changes. Just added a rubocop linter to the repository to make it easier to maintain a consistent code style. Could you trigger a change in your PR to run a linter and see results? If you have nothing to change you can just "touch" last commit by running `git commit --amend` and push the same commit with new sha `git push --force-with-lease`

Thanks for your time.

**Comment by hobbypunk90** [comment]
i understand the time problem 😁 
i fixed my linter problems 😊 but there are 2 more 😉

File: lib/omniauth/strategies/openid_connect.rb [hunk 2]
```diff
@@ -335,10 +337,10 @@ def verify_id_token!(id_token)
       class CallbackError < StandardError
         attr_accessor :error, :error_reason, :error_uri
 
-        def initialize(error, error_reason = nil, error_uri = nil)
-          self.error = error
-          self.error_reason = error_reason
-          self.error_uri = error_uri
+        def initialize(data)
```
Comment (m0n9oose) [inline lib/omniauth/strategies/openid_connect.rb]: Why do you prefer to use `Hash` instead of keyword arguments?

```ruby
def initialize(error:, reason: nil, uri: nil)
  self.error = error
  self.error_reason = reason
  self.error_uri = uri
end
```

this way you can validate the names of passed values and make sure there's mandatory `error` argument

File: lib/omniauth/strategies/openid_connect.rb [hunk 3]
```diff
@@ -126,8 +126,10 @@ def callback_phase
         client.authorization_code = authorization_code
         access_token
         super
-      rescue CallbackError, ::Rack::OAuth2::Client::Error => e
-        fail!(:invalid_credentials, e)
+      rescue CallbackError => e
+        fail!(e.error, e)
+      rescue ::Rack::OAuth2::Client::Error => e
+        fail!(e.response[:error], e)
```
Comment (m0n9oose) [inline lib/omniauth/strategies/openid_connect.rb]: Are sure sure this code works fine with new `data` argument? As far as I see from changes below you will get `ArgumentError` here because of number arguments (2 instead of 1) and I'm pretty sure `e.response[:error]` does not respond to `[]` method.

Comment (hobbypunk90) [reply lib/omniauth/strategies/openid_connect.rb]: I'm not sure, what you mean 🤔 
the second rescue is a complete different error class, my changes don't touch this part.
the first rescue works😁 we use it, as it is in a production system 😅

File: lib/omniauth/strategies/openid_connect.rb (hunk 2 above)
Comment (hobbypunk90) [reply lib/omniauth/strategies/openid_connect.rb]: the problem here ist, that we need a single param to construct the error.
With multiple params the `raise CallbackError, error: "...", reason: "..."` don't work.
